### ISSUE TITLE
Generate correct Eclipse execution environment for Java 9

### ISSUE THREAD

**Issue by vlotarev** [gradle/gradle#3046]
Everywhere in Gradle version number for Java 9 is **1.9** (got from JavaVersion enumeration) though should be just **9** (version scheme was changed). This produces problems in Eclipse plugin, for example. By default Gradle generates wrong Eclipse artifacts.

**Comment by blindpirate** [comment]
Related to #3010 .

**Comment by blindpirate** [comment]
@vlotarev Could you please provide us with some examples to reproduce the issue so that we could test our new version mechanism after we fix this?

**Comment by vlotarev** [comment]
It's easy. For example, apply eclipse plugin and execute eclipse task.  Than look into .classpath - you should see entry like
`<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-9/"/>`

Currently `JavaSE-1.9` is generated.

Also you may look into .settings/org.eclipse.jdt.core.prefs file. All properties like `org.eclipse.jdt.core.compiler.compliance` and `org.eclipse.jdt.core.compiler.source` should have value 9 but not 1.9 as currently.

**Comment by blindpirate** [comment]
@vlotarev Thanks!

**Comment by oehme** [comment]
@blindpirate this is unrelated to #3010. This is about the naming scheme of so-called "execution environments" that Eclipse uses.

**Comment by oehme** [comment]
@donat could you fix this before 4.3 RC1 early next week? Currently importing Java 9 projects into Eclipse fails because of this. The name of the execution environment doesn't match the pattern we use anymore.

**Comment by donat** [comment]
Sure, I'll do it.

**Comment by Tapter** [comment]
Is this fixed? gradle 4.3.1 still produces wrong versions in org.eclipse.jdt.core.prefs:
```
org.eclipse.jdt.core.compiler.source=1.9
org.eclipse.jdt.core.compiler.compliance=1.9
```
instead of
```
org.eclipse.jdt.core.compiler.source=9
org.eclipse.jdt.core.compiler.compliance=9
```

**Comment by donat** [comment]
This is a different issue. The fix you are looking for is part of Gradle 4.4: https://github.com/gradle/gradle/issues/3376
